### ISSUE TITLE
BadAtom (invalid Atom parameter)

### ISSUE THREAD

**Issue by mcmonkey4eva** [opentk/opentk#408]
Using latest OpenTK from a fresh GitHub release build, I get:

```
X Error of failed request: BadAtom (invalid Atom parameter)
```

With opcode 17 (X_GetAtomName), id 0x0, serial number 240.

This occurs whenever I click on the window.

This is on XUbuntu 16.04, in a GameWindow-using application running OpenGL 4.3 ForwardCompatible.

EDIT: This occurs on Ubuntu 16.04 (standard Unity desktop) as well :/

Due to the fact that this occurs consistently across my entire Linux testing environment and makes it unusable, I'd call this priority.

**Comment by varon** [comment]
Thanks for the report, @mcmonkey4eva.

That's really strange. Could you give us some more information about your drivers, graphics hardware, and mono version? Do you receive this error on the old NuGet package?

 Could you post a minimal example of the code triggering the issue?

**Comment by mcmonkey4eva** [comment]
- Non-replicable on Windows
- Only tested on the one machine currently
- Happens on mouse click
- No idea what part of my code triggers it, as I can't manage to replicate it with a short copy pasta of general window handling code (and mouse event console output)
- OS is Ubuntu 16.04 (64 bit)
- Graphics card is Nvidia GTX 860M, driver version: 364.19
- Is a laptop
- nvidia-settings reports: X Server version 11.0, vendor string x.org foundation, vendor version 1.18.3
- Single 1080p screen
- OpenGL version available: 4.50
- OpenGL version used in program: 4.30 (Core profile)
- Ubuntu generally updated last: in the middle of testing to check for any relation there
- Other technical notes that may be of relevance: Harddrive usage asks for a password once in a while (EG on startup), but there's no password and just hitting enter fixes it, but I could see it crashing from some internal harddrive read being refused.

I really don't know what code might be relevant to this. Is there anything I could look for and attempt to edit / copy to a temp project / etc. to test for relevance?

**Comment by cra0zy** [comment]
Are you just running an empty window when that happens? If not can you post a code? Also do google that error and see if it's maybe related to some other stuff you might be doing with X11.

**Comment by mcmonkey4eva** [comment]
Nope, it's a window filled with content.

`If not can you post a code?` Like, source code? It's a 50k+ lines of code project that suddenly stopped working on Linux with an OpenTK update. I doubt you'd want to dredge through that. If you mean the error codes: all in the opening post.
`Also do google that error` I did, all I can decipher is that it's some obscure X11 feature that I assume OpenTK uses somewhere.
`you might be doing with X11.` I don't interact with X11 directly, I use OpenTK for wrapping that stuff into a neat cross-platform setup.

**Comment by cra0zy** [comment]
> If not can you post a code? Like, source code? It's a 50k+ lines of code project that suddenly stopped working on Linux with an OpenTK update. I doubt you'd want to dredge through that. If you mean the error codes: all in the opening post.

I meant more like a small piece of code which can reproduce the bug.

**Comment by mcmonkey4eva** [comment]
@cra0zy That's the problem: I have absolutely no idea what is causing this, so I have no idea what piece of code I would copy for that. It happens in my massive project, it doesn't happen in a quick miniproject with matched window settings

**Comment by varon** [comment]
Thanks for the detailed feedback, @mcmonkey4eva.
If you can't reproduce it with a minimal example, do you think you could post a stack trace?

**Comment by mcmonkey4eva** [comment]
@varon I would love to post a stacktrace... this does not produce a stacktrace though. I could screenshot the error or something, but it'd be the same information as is available in the opening post.
I'd imagine this is a C-code level error, (meaning no exceptions, it just dumps what it can to console and `exit(1);`)

**Comment by varon** [comment]
@mcmonkey4eva That's going to make it _really_ tough.

It's probably some wierd API usage by OpenTK behind the scenes, probably triggered by the specific setup or execution order in your project. (It's likely an OpenTK issue that so far hadn't been triggered, so we're happy to continue trying to help fix it.

Here's my suggested approach to getting a repro:

Copy your project, build and check you still receive the error. Then, remove some code. If the errors stops, re-add the code. If it doesn't, continue removing code until it does. Ultimately you should end up with a working minimal repro example.

**Comment by mcmonkey4eva** [comment]
Update:

This is rather annoying.
I reduced my code to a single file that replicated it.
Then I copied the replication code to a new project... and it didn't replicate anymore.
Something's wrong with... the project config files somehow?

here's full and short replication files if you want to boot up Ubuntu 16.04 and see if you can get it happening on your own machine(s), or just look through the code and project config. https://dl.dropboxusercontent.com/u/54130586/v4.zip

**Comment by varon** [comment]
@mcmonkey4eva Thanks! It should really help having an example to work from.

**Comment by varon** [comment]
@mcmonkey4eva  - Did we ever figure out what caused this?

Can we close?

**Comment by varon** [comment]
Closing, just re-open if it's still an issue.

**Comment by mcmonkey4eva** [comment]
Haven't the slightest because it's been so long. Might as well close for now until/unless a repeat of the issue appears.

**Comment by varon** [comment]
Thanks for checking in @mcmonkey4eva. Appreciated.

**Comment by UnknownShadow200** [comment]
Dropbox link 404s now.

**Comment by danielbisar** [comment]
We have the same issue in one of our projects. Am going to do some analysis and try to give more detailed feedback in the next days.

**Comment by varon** [comment]
@danielbisar - Keep us posted!

**Comment by jvbsl** [comment]
https://www.mono-project.com/docs/debug+profile/debug/#debugging-with-gdb
Perhaps that helps if you can't do step debugging or get a StackTrace.
Apart from that it should be fixed with 4.0. if it is not probably glfws fault^^

**Comment by danielbisar** [comment]
Sorry for not coming back earlier. Since the problem was solved (seemed to be caused by not installed vbox tools in virtual box) and other projects have higher priority for us right now, I am sorry. If I will find out more or have some spare time I will come back.
